ts/main.ts

```diff
@@ -0,0 +1,140 @@
+if (!$weatherDataDiv) throw new Error('weather data div query failed');
+if (!$form) throw new Error('form query failed');
+if (!$locationInput) throw new Error('location input query failed');
+if (!$cityName) throw new Error('city name query failed');
+if (!$weatherDescription) throw new Error('weather description query failed');
+if (!$weatherImg) throw new Error('weather image query failed');
+if (!$temperature) throw new Error('temperature query failed');
+if (!$humidity) throw new Error('humidity query failed');
+if (!$windSpeed) throw new Error('wind speed query failed');
+
+const apiKey = 'b03d66ceebc0a945c3eb2bb9cc3551d1';
+
+$form.addEventListener('submit', async (event: Event) => {
+  event.preventDefault();
+  const location = $locationInput.value;
+  await fetchData(location);
+});
+
+$weatherForm2.addEventListener('submit', async (event: Event) => {
+  event.preventDefault();
+  const location = $locationInput2.value;
+  await fetchData(location);
+});
+
+$backButton.addEventListener('click', () => {
+  swapView('weather-app');
+});
+
+async function fetchData(location: string): Promise<void> {
+  try {
+    const response = await fetch(
+      `https://api.openweathermap.org/data/2.5/weather?q=${location}&appid=${apiKey}&units=imperial`,
+    );
+    if (!response.ok) {
+      throw new Error('HTTP Error');
+    }
+    const data = await response.json();
+    displayWeatherData(data);
+    swapView('weather-display');
+    $locationInput.value = '';
+  } catch (error) {
+    console.log('Error:', error);
+  }
+}
+
+interface WeatherData {
+  weather: { description: string; main: string }[];
+  main: { temp: number; humidity: number };
+  wind: { speed: number };
+  name: string;
+}
+
+function displayWeatherData(data: WeatherData): void {
+  console.log(data);
+  let weatherImage = '';
+  $weatherDescription.textContent = data.weather[0].description;
+  const weatherCondition = data.weather[0].main.toLowerCase();
+  $cityName.textContent = data.name;
+  $weatherImg.alt = data.weather[0].description;
+
+  const roundedTemperature = Math.round(data.main.temp);
+  $temperature.textContent = `${roundedTemperature} °F`;
+
+  $humidity.textContent = `Humidity: ${data.main.humidity}%`;
+  $windSpeed.textContent = `Wind Speed: ${data.wind.speed} mph`;
+
+  switch (weatherCondition) {
+    case 'clear':
+      $background.className = 'bg-clear';
+      weatherImage = 'images/sunny.png';
+      break;
+    case 'clouds':
+    case 'haze':
+      $background.className = 'bg-clouds';
+      weatherImage = 'images/cloudy.png';
+      break;
+    case 'rain':
+    case 'drizzle':
+      $background.className = 'bg-rain';
+      weatherImage = 'images/rainy.png';
+      break;
+    case 'windy':
+      $background.className = '';
+      weatherImage = 'images/windy.png';
+      break;
+    default:
+      weatherImage = 'images/default.png';
+  }
+
+  $weatherImg.src = weatherImage;
+}
+
```

Comment:
You should also add `alt` attributes to your image. Something about what weather icon this is so...

`Icon of weather for ${weatherCondition} weather`

Then if the image breaks or someone uses a screen reader, it would say something like "Icon of weather for clear weather"